apps/aecore/src/aec_peers.erl

```diff
@@ -554,11 +556,22 @@ log_ping_and_set_reping(Res, CalcF, Uri, Time, State) ->
             State;
         {value, _Hash, Peer} ->
             update_ping_metrics(Res),
-            Peer1 = save_ping_event(Res, Time, Peer),
-            Peer2 = start_ping_timer(CalcF, Peer1),
-            Peers = enter_peer(Peer2, State#state.peers),
-            Errored = update_errored(Res, uri_of_peer(Peer), State#state.errored),
-            State#state{peers = Peers, errored = Errored}
+            %% drop an errored peer if last time seen is longer than expire interval ago
+            Expire = 
+               application:get_env(aecore, peer_error_expired, ?DEFAULT_EXPIRE_INTERVAL),
```

Comment:
perhaps `peer_error_expiry` would be more correct?

Reply:
Of course, that's a much better name. I rebased against master and changed it accordingly